ndmg/stats/qa_tensor.py

```diff
@@ -33,86 +34,167 @@
+                
+                #relies on the fact that peaks_from_models generates peak_dirs and peak_values in descending order
+                directions.append(peak_dirs[x,y,z,0,:])
+                heights.append(peak_values[x,y,z,0])
+
+    #convert to np arrays
+    centers = np.asarray(centers)
+    directions = np.asarray(directions)
+    heights = np.asarray(heights)
+    
+    #get rgb based on directional components
+    directions_colors = orient2rgb(directions)
+    
+    return centers,directions,directions_colors,heights
+
+def plot_directions(peak_dirs,peak_values,x_angle,y_angle,fname,size=(300,300)):
     """
-    data: fa map
+    Opens a 3-d fury window of the maximum peaks visualized
+    
+    To show a slice, provide a sliced volume of the peak_dirs and peak_values and adjust the x_angle and y_angle 
+    See Tractography Directional Field QA Tutorial for examples
+    
+    Parameters
+    -----------
+    peak_dirs: np array of peak_dirs 
+    peak_values: np array of peak_values
+    slices: dictionary of lists of slices in x,y,z plane
+    x_angle: angle to rotate image along x axis
+    y_angle: angle to rotate image along y axis
+    fname: str path to save
+    size: size tuple for each image
+    """    
+    centers,directions,directions_colors,heights = generate_3_d_directions(peak_dirs,peak_values)
+    
+    scene = window.Scene()
+    arrow_actor = actor.arrow(centers, directions, directions_colors, heights)
+
+    scene.add(arrow_actor)
+    scene.roll(x_angle)
+    scene.pitch(y_angle)
+    
+    window.show(scene, size = size)
+    
+def pad_im(image,max_dim,pad_val):
     """
-    im = data.get_data()
-    fig = plot_rgb(im)
-    fname = os.path.split(fname)[1].split(".")[0] + ".png"
-    plt.savefig(outdir + fname, format="png")
-
+    Pads an image to be same dimensions as given max_dim
+    
+    Parameters
+    -----------
+    image: 3-d RGB np array of image slice
+    max_dim: dimension to pad up to
+    pad_val: value to pad with
+    
+    Returns
+    -----------
+    padded_image: 3-d RGB np array of image slice with padding
+    """
+    #pad only in first two dimensions not in rgb
+    pad_width = (((max_dim-image.shape[0])//2,(max_dim-image.shape[0])//2),((max_dim-image.shape[1])//2,(max_dim-image.shape[1])//2),(0,0))
+    padded_image = np.pad(image, pad_width=pad_width, mode='constant', constant_values=pad_val)
+    
+    return padded_image
 
-def plot_rgb(im):
-    plt.rcParams.update({"axes.labelsize": "x-large", "axes.titlesize": "x-large"})
+def create_qa_figure(peak_dirs,peak_values,output_dir,model):
+    """
+    Creates a 9x9 figure of the 3-d volume and saves it
+    
+    Parameters
+    -----------
+    peak_dirs: np array of peak_dirs 
+    peak_values: np array of peak_values
+    output_dir: location to save qa figure
+    model: model type (CSA, CSD)
+    """
+    #set shape of image
+    im_shape = peak_dirs.shape[0:3]
+    im_shape_rgb = im_shape + (3,)
```

Comment:
if you're going to mutate a tuple, it should be a list, since tuples are immutable.
so you can just do
```
im_shape = list(peak_dirs.shape[:3])
im_shape_rgb = [*im_shape, 3]
```